lib/PuppeteerSharp/ChromeTargetManager.cs

```diff
@@ -182,9 +194,10 @@ private void OnTargetCreated(TargetCreatedResponse e)
             }
         }
 
-        private void OnTargetDestroyed(TargetDestroyedResponse e)
+        private async void OnTargetDestroyed(TargetDestroyedResponse e)
         {
             _discoveredTargetsByTargetId.TryRemove(e.TargetId, out var targetInfo);
+            await _targetDiscoveryCompletionSource.Task.ConfigureAwait(false);
```

Comment:
Although we are trying to enforce this. What do you think about using `WithTimeout`, and throwing an exception? If we fail to complete this task somehow, the code will get stuck here.

Reply:
Note that `_targetDiscoveryCompletionSource` will be set if and only if the message `Target.setDiscoverTargets` completes. I can't think of any scenario where it wouldn't complete.

There's no code that uses a timeout value in ChromeTargetManager. Where should I derive the timeout value from?

Reply:
Let's use `Puppeteer.DefaultTimeout`. It's quite a long timeout. We want to prevent the code from hanging.